fili-core/src/main/java/com/yahoo/bard/webservice/data/config/provider/ConfiguredTableLoader.java

```diff
@@ -0,0 +1,111 @@
+import com.yahoo.bard.webservice.data.config.table.PhysicalTableDefinition;
+import com.yahoo.bard.webservice.data.time.TimeGrain;
+import com.yahoo.bard.webservice.table.TableGroup;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A TableLoader from configuration.
+ */
+public class ConfiguredTableLoader extends BaseTableLoader {
+
+    protected final ConfigurationDictionary<LogicalTableConfiguration> logicalTables;
+    protected final ConfigurationDictionary<PhysicalTableConfiguration> physicalTables;
+    protected final ConfigurationDictionary<DimensionConfig> dimensions;
+
+    /**
+     * Construct a table loader from configuration.
+     *
+     * @param logicalTables the logical tables
+     * @param physicalTables the physical tables
+     * @param dimensions the dimensions
+     */
+    public ConfiguredTableLoader(
+            ConfigurationDictionary<LogicalTableConfiguration> logicalTables,
+            ConfigurationDictionary<PhysicalTableConfiguration> physicalTables,
+            ConfigurationDictionary<DimensionConfig> dimensions
+    ) {
+        this.logicalTables = logicalTables;
+        this.physicalTables = physicalTables;
+        this.dimensions = dimensions;
+    }
+
+    @Override
+    public void loadTableDictionary(ResourceDictionaries dictionaries) {
+
+        for (Map.Entry<String, LogicalTableConfiguration> entry : logicalTables.entrySet()) {
+            LogicalTableConfiguration logicalTable = entry.getValue();
+            String tableName = entry.getKey();
+
+            // Compute the set of all physical fields on this logical table's physical tables
+            Set<FieldName> physicalFields = logicalTable
+                    .getPhysicalTables()
+                    .stream()
+                    .map(physicalTable -> physicalTables.get(physicalTable).getMetrics())
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toSet());
+
+            // Build the actual physical tables for this logical table
+            Set<PhysicalTableDefinition> physicalTableDefs = logicalTable
+                    .getPhysicalTables()
+                    .stream()
+                    .map(physicalTable -> physicalTables.get(physicalTable).buildPhysicalTable(dimensions))
+                    .collect(Collectors.toSet());
+
+            // Get the smallest granularity for this table
+            // TimeGrain smallestGrain = Collections.min(entry.getValue().timeGrains);
+
+            // Fixme: We should use 'apiName' vs 'asName' as intended
+            // FIXME: should be possible to override grains valid for?
+            Set<ApiMetricName> apiMetricNames = new HashSet<>();
+            for (String metric : logicalTable.getMetrics()) {
+                apiMetricNames.add(new ApiMetricName() {
+                    @Override
```

Comment:
It might make sense to extract this ApiMetricName factory code into a method so as to make it more easily tested.  It's also not strictly necessary to give every ApiMetricName it's own anonymous class.